[W13-4] giatros

wfxronald opened the pull request:
@chuaes @sickerin

ewaldhew commented on README.adoc:
```diff
@@ -1,10 +1,9 @@
-https://www.codacy.com/app/damith/addressbook-level4?utm_source=github.com&utm_medium=referral&utm_content=se-edu/addressbook-level4&utm_campaign=Badge_Grade[image:https://api.codacy.com/project/badge/Grade/fc0b7775cf7f4fdeaf08776f3d8e364a[Codacy Badge]]
+https://travis-ci.org/CS2103-AY1819S2-W13-4/main[image:https://travis-ci.org/CS2103-AY1819S2-W13-4/main.svg?branch=master[Build Status]]
+https://ci.appveyor.com/project/chuaes/main-erei4[image:https://ci.appveyor.com/api/projects/status/l0jp6ewmhauhmsr0?svg=true[Build status]]
+https://coveralls.io/github/CS2103-AY1819S2-W13-4/main?branch=master[image:https://coveralls.io/repos/CS2103-AY1819S2-W13-4/main/badge.svg?branch=master[Coverage Status]]
```
Good updating all these :)

ewaldhew commented on README.adoc:
```diff
@@ -14,10 +13,8 @@ ifndef::env-github[]
-* It is *written in OOP fashion*. It provides a *reasonably well-written* code example that is *significantly bigger* (around 6 KLoC)than what students usually write in beginner-level SE modules.
-* What's different from https://github.com/se-edu/addressbook-level3[level 3]:
+* This is a desktop Giatros application. It has a GUI but most of the user interactions happen using a CLI (Command Line Interface).
+* It is a Java sample application intended for users(patients, receptionist, doctors and pharmacist).
```
more description of the product would be nice

ewaldhew commented on README.adoc:
```diff
@@ -32,8 +29,7 @@ endif::[]
 
-* Some parts of this sample application were inspired by the excellent http://code.makery.ch/library/javafx-8-tutorial/[Java FX tutorial] by
-_Marco Jakob_.
+* Some parts of this sample application were inspired by https://github.com/nus-cs2103-AY1819S1/AddressBook-level4[AddressBook4].
```
Add another line, not replace the existing 3rd party credits

ewaldhew commented on docs/AboutUs.adoc:
```diff
@@ -4,53 +4,44 @@
 {empty} +
 We are a team based in the http://www.comp.nus.edu.sg[School of Computing, National University of Singapore].
 
 == Project Team
```
Roles: Who is in charge of which component?

ewaldhew commented on docs/UserGuide.adoc:
```diff
@@ -12,19 +12,22 @@ ifdef::env-github[]
+Giatros is a hospital-management app that facilitate better interactions between different parties involved in the hospital.
+Mainly targeted at the hospital receptionists who are keeping track of the patient records, Giatros can also be used by doctors
+who wish to find out more about their patients, allows patients to manage their appointments and also by pharmacists
+who wish to prescribe a more personalised medication to the patient.
```
Good

ewaldhew reviewed (commented):
Remember to update the project description on Github! The line on top that says "Address Book sample application"

Fix user and dev guides based on the previous comments also (in the pdf)

ewaldhew commented on docs/DeveloperGuide.adoc:
```diff
@@ -827,35 +836,86 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+1. Receptionist encounters a new patient.
+2. Receptionist checks whether the new patient is registered in the system.
+3. Receptionist requests for the patient’s personal data.
+4. Receptionist adds the patient into the system.
```
Use cases should alternate action between user and system!

chuaes committed:
fix RedundantImport and Linelength issues

chuaes committed:
fix lineLength for addCommand sys test

chuaes committed:
Merge branch 'master' into master

chuaes committed:
Merge pull request #63 from chuaes/master

Refactoring of role (person to patient)

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-W13-4/main into v1.2

# Conflicts:
#	src/main/java/seedu/giatros/logic/commands/AddallCommand.java
#	src/main/java/seedu/giatros/logic/commands/EditCommand.java
#	src/main/java/seedu/giatros/logic/parser/AddCommandParser.java
#	src/main/java/seedu/giatros/model/patient/Patient.java
#	src/main/java/seedu/giatros/model/util/SampleDataUtil.java
#	src/main/java/seedu/giatros/storage/JsonAdaptedPerson.java
#	src/test/java/seedu/giatros/logic/commands/AddallCommandTest.java
#	src/test/java/seedu/giatros/logic/parser/GiatrosBookParserTest.java
#	src/test/java/seedu/giatros/testutil/PersonBuilder.java
#	src/test/java/seedu/giatros/ui/PersonListPanelTest.java

Commit:
Refactor 'Person' to 'Patient'
- Due to conflicts, some changes need to be committed again separately

Commit:
Refactor a few more files: 'Person' to 'Patient'

Commit:
Rename JsonAdaptedPerson to JsonAdaptedPatient

Commit:
Add 'Allergy' field to JsonAdaptedPatient class

Commit:
Add withAllergy() for PatientBuilder

Commit:
Connect Allergy field to PatientCard
- Modify the PatientCard's constructor to add the allergy
- Modify GuiTestAssert to compare the allergy label

Commit:
Edit the logic of 'addall' and update tests for execute

Commit:
Add EOF for giatrosbook.json

Commit:
Remove checkStyle error in AddallCommandTest

Commit:
Remove more checkStyle error

Commit:
Remove minor checkStyle error - unused import

sickerin committed:
Merge pull request #62 from wfxronald/v1.2

V1.2 - Continuing to work on 'addall' command

wfxronald committed:
Configure reposense

wfxronald committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-W13-4/main into v1.2

wfxronald committed:
Undo reposense configuration

wfxronald committed:
Refactor Tag as Allergy to take in multiple allergies
- Currently, the old Allergy feature is still inside

wfxronald committed:
Edit checkStyle error in SampleDataUtil.java

wfxronald committed:
Edit to remove checkStyle error

chuaes committed:
Merge pull request #70 from wfxronald/ver1.3

Allow multiple allergies to be added: Closes #66

chuaes committed:
Merge pull request #1 from CS2103-AY1819S2-W13-4/master

Allergy tag being added

chuaes committed:
register and add account

chuaes committed:
Merge remote-tracking branch 'origin/master' into chuaesv1.2

wfxronald committed:
Update the various documentations

wfxronald committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-W13-4/main into ver1.3

chuaes committed:
Merge pull request #2 from chuaes/chuaesv1.2

registering and login functionality added

chuaes committed:
filled up the json file

chuaes committed:
Merge pull request #3 from chuaes/chuaesv1.2

filled up the json file

chuaes committed:
Fixed all the checkstyle issues

chuaes committed:
Merge pull request #4 from chuaes/chuaesv1.2

Fixed all the checkstyle issues

chuaes committed:
Fix checkstyle issue

chuaes committed:
Merge pull request #5 from chuaes/chuaesv1.2

Fix checkstyle issue

chuaes committed:
fix checkstyle issue

chuaes committed:
Merge pull request #6 from chuaes/chuaesv1.2

fix checkstyle issue

wfxronald committed:
Update the app to version 1.3.0

chuaes committed:
fix checkstyle issue

chuaes committed:
Merge pull request #7 from chuaes/chuaesv1.2

fix checkstyle issue

wfxronald committed:
Merge pull request #73 from wfxronald/ver1.3

Update the various documentations

chuaes committed:
Merge branch 'master' into master

chuaes committed:
Merge pull request #75 from chuaes/master

Registering and login functionality added

wfxronald committed:
Integrate addAll command to Allergy tag

chuaes committed:
Merge pull request #79 from wfxronald/ver1.4

Integrate addAll command to Allergy tag. Closes #68

wfxronald committed:
Update the json test file with accounts
- Previously tests were failing because new login feature was added, expecting a new json format

wfxronald committed:
Creating a new remall command
- This behaves in the exact opposite manner of addall command
- Tests have not been added yet

wfxronald committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-W13-4/main into ver1.4

wfxronald committed:
Add line at EOF for giatrosbook.json

wfxronald committed:
Remove checkStyle error due to if-else braces

wfxronald committed:
Remove checkStyle error due to unused import

wfxronald committed:
Merge pull request #81 from wfxronald/ver1.4

Creation of a new remall command

wfxronald committed:
Update the various documentations and first draft PPP

wfxronald committed:
Some UI improvements

wfxronald committed:
Update Javadoc comment to tackle checkStyle error

sickerin committed:
Merge remote-tracking branch 'upstream/master'

wfxronald committed:
Edit minor error due to search keyword bound by brackets
- Also updated the failing tests

wfxronald committed:
Update to green theme

chuaes committed:
Merge pull request #83 from wfxronald/ui

Some GUI improvements. Closes #56

chuaes committed:
login, logout and register features enhancement and bug fixes

chuaes committed:
Logout feature

chuaes committed:
gui update

chuaes committed:
default image for brower panel holder

chuaes committed:
Checkstyle fixes

chuaes committed:
checkstyle fixes

chuaes committed:
checkstyle fix

wfxronald committed:
Merge branch 'master' into docs

wfxronald committed:
Merge branch 'master' of https://github.com/wfxronald/main into docs

wfxronald committed:
Merge branch 'docs' of https://github.com/wfxronald/main into docs

chuaes committed:
account creator class is used for testing and bug fix for before-login constraints

chuaes committed:
Checkstyle fixes

chuaes committed:
checkstyle fixes

wfxronald committed:
Merge pull request #85 from chuaes/master

Bug fix for login feature, new logout feature

chuaes committed:
Merge branch 'master' into docs

chuaes committed:
Merge pull request #82 from wfxronald/docs

Updating the various documentations. Closes #76

chuaes committed:
Add "head staff" role and put more restriction to normal staff role

chuaes committed:
Merge branch 'master' into master

chuaes committed:
checkstyle fixes

chuaes committed:
Merge branch 'master' of https://github.com/chuaes/main

wfxronald committed:
Merge pull request #86 from chuaes/master

Add "head staff" role and put more restrictions to normal staff role

sickerin committed:
Merge remote-tracking branch 'upstream/master'

sickerin committed:
Merge remote-tracking branch 'origin/master'

radhika-lakhtakia committed:
Merge pull request #1 from CS2103-AY1819S2-W13-4/master

Pull!!!!

wfxronald committed:
Update the tests for remall and others

sickerin committed:
add appointment class

class is incomplete, regex not filled out

sickerin committed:
add AddaptCommandParser

need to add parser still....

sickerin committed:
add AddAptCommand

refactored version, need to make it suitable

wfxronald committed:
Fix the build.gradle file to run test properly

sickerin committed:
modified Patient and AddAptCommand

add Prefix to CliSyntax

sickerin committed:
Merge pull request #89 from wfxronald/docs

Update the tests for remall and others. Closes #77

wfxronald committed:
Edit typo in user guide to reflect the tag addremall

wfxronald committed:
Refactor addall and remall command
- Ensure that not removing/adding allergy is considered an exception, and should not modify the giatrosbook state (not undo/redoable)
- Change the test to assert failure instead
- Change the instruction for addall and remall to reflect the user guide
- Change the error handling for addall and remall when only index is provided but no allergy tags

wfxronald committed:
Change the app version

wfxronald committed:
Remove unused import in RemallCommandTest

wfxronald committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-W13-4/main into docs

sickerin committed:
add appointment to ParserUtil, corrected typos

untested

sickerin committed:
add AddaptCommand to GiatrosBookParser

add example in addaptcommand, corrected bucks in addallcommand

sickerin committed:
modify EditCommand for appointment

sickerin committed:
modified SampleDataUtil and RemallCommand to account for appointments

sickerin committed:
fixed typo in SampleDataUtil

chuaes committed:
Merge pull request #90 from wfxronald/docs

Refactoring addall and remall and other minor changes

chuaes committed:
Merge pull request #1 from CS2103-AY1819S2-W13-4/master

bug fixed in build.gradle file and minor changes related to allergy feature

sickerin committed:
add JsonAdapted versions; fixed typos and modified parsers for apt

Working well

chuaes committed:
Fixes to wrong message displayed after entering a command

chuaes committed:
Fix checkstyle issues

chuaes committed:
Merge pull request #92 from chuaes/master

 Fix wrong message displayed issue after entering a command

wfxronald committed:
Update the documentation diagrams in developer guide

wfxronald committed:
Merge branch 'master' into docs

sickerin committed:
add appointments to UI; refactor and modify UI files

wfxronald committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-W13-4/main into docs

wfxronald committed:
Merge branch 'docs' of https://github.com/wfxronald/main into docs

sickerin committed:
Merge pull request #93 from wfxronald/docs

Updating developer guide and diagrams

sickerin committed:
add appointments unit test and  mod misc files for appointments

only 2 tests failing

sickerin committed:
add appointments .... , modify giatrosbook.json (see prev commit)

prev commit was a mistake

sickerin committed:
gitignore update

sickerin committed:
configure so that all tests pass locally (see description)

- add and corrected apt related prefixes across repo
- remove allergy from patient requirenonnull
- account for appointment in parser tests
- added space for descriptor builders when apt/ or y/ is null so that we don't get y/apt
- modified typical patients to account for appointments
- account for appointment in systemtests (add and edit)

sickerin committed:
Merge remote-tracking branch 'upstream/master'

sickerin committed:
resolve merge conflicts, modify RemallCommandTest for appointment

still one error in remallcommandtest

sickerin committed:
Update UserGuide.adoc

sickerin committed:
fix checkstyleMain

sickerin committed:
fix checkstyleTest

sickerin committed:
Update UserGuide.adoc

sickerin committed:
fix mistake in RemallCommandTests, upload sickerin.adoc

sickerin committed:
Merge remote-tracking branch 'origin/master'

sickerin committed:
remove system.out.print; fix typo in AddCommandParser javadocs

wfxronald committed:
Edit typo in documentation for add/remall

sickerin committed:
fix EOF and trailing whitespace

sickerin committed:
some changes to sickerin.adoc

sickerin committed:
Merge pull request #96 from sickerin/master

add appointment functionality and modify files to account for it

sickerin committed:
Merge branch 'master' into docs

wfxronald committed:
Update user guide: new UI screenshot and authentication part

wfxronald committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-W13-4/main into docs

wfxronald committed:
Merge branch 'docs' of https://github.com/wfxronald/main into docs

wfxronald committed:
Minor PPP update

sickerin committed:
add AddaptCommandTest, modify AddaptCommand for duplicate error

sickerin committed:
fix appointment tests for GiatrosBookParser and AddaptCommand;

fix usersession for addaptcommandparser

chuaes committed:
Merge pull request #2 from CS2103-AY1819S2-W13-4/master

Add appointment feature

sickerin committed:
add Remapt command and associated parsers

sickerin committed:
add RemaptCommand related tests

- modified typical patients idx 1 with appointment to avoid significant refactoring of tests
- modified RemallCommandTest with appointments
- add appointments to PatientCardHandle

wfxronald committed:
Update developer guide: manual test, add/remall

chuaes committed:
Tests added to increase code coverage

chuaes committed:
Checkstyle fix

chuaes committed:
checkstyle fix

chuaes committed:
Checkstyle fix

sickerin committed:
fix bug no null pointer exception for Appointment

sickerin committed:
Merge remote-tracking branch 'upstream/master'

sickerin committed:
fix Checkstylemain and test, add remapt to GiatrosBookParserTest

sickerin committed:
fix StyleCheck

sickerin committed:
Merge pull request #97 from wfxronald/docs

Updating documentations

chuaes committed:
Checkstyle fixes

chuaes committed:
Checkstyle fixes

wfxronald committed:
Update developer guide: remall and design choice

chuaes committed:
Fixed checkstyle

chuaes committed:
Fix checkstyle

chuaes committed:
fix checkstyle

chuaes committed:
Fix checkstyle

chuaes committed:
checkstyle fix

chuaes committed:
Checkstyle fixes

chuaes committed:
Checkstyle fixes

chuaes committed:
checkstyle fixes

chuaes committed:
fix checkstyle

sickerin committed:
Update RegisterCommandParserTest.java

fix checkstyleTest

sickerin committed:
Merge remote-tracking branch 'upstream/master'

chuaes committed:
Merge branch 'master' into master

chuaes committed:
Update register related class to fix checkstyle issue

chuaes committed:
fix checkstyle

chuaes committed:
Merge branch 'master' into master

sickerin committed:
Merge remote-tracking branch 'origin/master'

chuaes committed:
fix login test

chuaes committed:
Merge branch 'master' of https://github.com/chuaes/main

chuaes committed:
Remove uncessary test from login

sickerin committed:
update user docs and developerguide for appointments

sickerin committed:
Merge pull request #98 from chuaes/master

Tests added to increase code coverage

sickerin committed:
Merge branch 'master' into upstream/master

sickerin committed:
Merge remote-tracking branch 'upstream/master'

sickerin committed:
Merge remote-tracking branch 'upstream/master'

sickerin committed:
import addapt to logicmanager

wfxronald committed:
Merge pull request #99 from sickerin/master

Add remove appointment command and associated unit tests

wfxronald committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S2-W13-4/main into docs

wfxronald committed:
More test for CommandResultTest

wfxronald committed:
Fixing documentation tags, remove template PPP

radhika-lakhtakia committed:
Merge pull request #2 from CS2103-AY1819S2-W13-4/master

Pull new changes

wfxronald committed:
Update about us page

radhika-lakhtakia committed:
Export to csv command + test

chuaes committed:
Update the ug to reflect the current product

sickerin committed:
fix typo in addaptCommand; update DeveloperGuide

wfxronald committed:
Edit powerpoint for diagrams and update png diagrams

sickerin committed:
updateDeveloperGuide

sickerin committed:
updateDeveloperGuide

sickerin committed:
remove misc comments

sickerin committed:
fix EOF and Developer Guide

wfxronald committed:
Merge branch 'master' into master

sickerin committed:
Merge pull request #100 from wfxronald/docs

Final edit to the product

sickerin committed:
Merge remote-tracking branch 'upstream/master'

chuaes committed:
Add more restrictions to managerial role

chuaes committed:
Merge branch 'master' of https://github.com/chuaes/main

chuaes committed:
Fix checkstyle

chuaes committed:
Merge branch 'master' into master

wfxronald committed:
Update PPP

chuaes committed:
Fix failed test during to new manager's constraint

wfxronald committed:
Remove remnants of Allergy and update about us

chuaes committed:
whatever

chuaes committed:
Merge branch 'master' of https://github.com/chuaes/main

sickerin committed:
updated sickerin.adoc

refactor person to patients
allegied to allergies
updated

sickerin committed:
fix Whitespace adoccheckstyle

chuaes committed:
Remove failed test

chuaes committed:
Update doc and diagram to reflect the new changes

chuaes committed:
Merge pull request #105 from wfxronald/docs

Final clean-up

chuaes committed:
Merge pull request #3 from CS2103-AY1819S2-W13-4/master

Update doc to reflect new changes

sickerin committed:
Merge branch 'master' into master

sickerin committed:
Merge remote-tracking branch 'upstream/master'

sickerin committed:
Auto stash before merge of "master" and "upstream/master"

sickerin committed:
Merge remote-tracking branch 'origin/master'

sickerin committed:
fix trailing whitespace

chuaes committed:
Fix checkstyle

radhika-lakhtakia committed:
Export tests added

chuaes committed:
Merge pull request #103 from sickerin/master

Update Developer Guide

chuaes committed:
Merge branch 'master' into master

wfxronald committed:
Update README.adoc

radhika-lakhtakia committed:
Checkstyle

wfxronald committed:
Last minute DG change

chuaes committed:
Update the dg

chuaes committed:
Merge pull request #104 from chuaes/master

Add more restrictions to manager account

chuaes committed:
Merge branch 'master' of https://github.com/chuaes/main

wfxronald committed:
Update README.adoc

sickerin committed:
Merge branch 'master' into master

sickerin committed:
Merge branch 'master' into wfxronald-patch-1

wfxronald committed:
Merge branch 'master' into docs

chuaes committed:
Merge pull request #106 from CS2103-AY1819S2-W13-4/wfxronald-patch-1

Update README.adoc

chuaes committed:
Merge pull request #108 from wfxronald/docs

Last minute DG change

chuaes committed:
Merge pull request #107 from chuaes/master

Update the dg to reflect the user authentication feature

radhika-lakhtakia committed:
Docs updated

radhika-lakhtakia committed:
Merge branch 'master' into master

radhika-lakhtakia committed:
Update docs

radhika-lakhtakia committed:
Merge branch 'docsb'

radhika-lakhtakia committed:
Update docs pt 2

radhika-lakhtakia committed:
stop opening csv

radhika-lakhtakia committed:
:)

radhika-lakhtakia committed:
whitespace

sickerin committed:
Merge pull request #101 from radhika-lakhtakia/master

Export Command

radhika-lakhtakia committed:
Export Tests

radhika-lakhtakia committed:
Fixed small discrepancy

radhika-lakhtakia committed:
fixed test case

radhika-lakhtakia committed:
Small Error Fixed

radhika-lakhtakia committed:
Fixed jar error? (Added line after filewrite)

radhika-lakhtakia committed:
Jar fixed

radhika-lakhtakia committed:
Jar fixed

wfxronald committed:
Merge branch 'master' into master

sickerin committed:
Merge pull request #109 from radhika-lakhtakia/master

Export Changes
